File: src/main/java/io/github/dqualizer/dqlang/types/rqa_definition/Parametrization.java
package io.github.dqualizer.dqlang.types.rqa_definition;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.ToString;

import java.util.Map;

@Getter
@ToString
public class Parametrization {

    @JsonProperty("path_variables")
    private Map<String, String> pathVariables;
    @JsonProperty("url_parameter")
    private Map<String, String> urlParameter;
    @JsonProperty("request_parameter")
    private Map<String, String> requestParameter;
    private String payload;
}

File: src/main/kotlin/dqualizer/dqlang/archive/loadtesttranslator/dqlang/modeling/Parametrization.kt
package dqualizer.dqlang.archive.loadtesttranslator.dqlang.modeling

import com.fasterxml.jackson.annotation.JsonProperty
import lombok.Getter
import lombok.ToString

@Getter
@ToString
class Parametrization {
    @JsonProperty("path_variables")
    private val pathVariables: Map<String, String>? = null

    @JsonProperty("url_parameter")
    private val urlParameter: Map<String, String>? = null //Size should be 1

    @JsonProperty("request_parameter")
    private val requestParameter: Map<String, String>? = null //Size should be 1
    private val payload: Map<String, String>? = null //Size should be 1
}
